### DISCUSSION TITLE
Problem accessing Slinger with dynamic DNS name

### DISCUSSION THREAD

**Discussion by tvpat** [GerryDazoo/Slinger#144] [Q&A] [answered]
Gerry,

First of all, thank you for doing this project.  I use my Slingbox often when I travel, and there is no replacement for it.

I have got Slinger working with my external home IP address but it is not working well with a dynamic dns name.

I set up a Windows 11 laptop to run slingbox_server.exe in the directory C:\slingbox on startup.  I have forwarded ports 8080 and 8081 to that laptop.  I have also forwarded ports 5301 and 5001 to their respective Slingboxes, but I don't think I need that.  My config.ini is below.

In my examples I have replaced my slingbox passwords, external home IP address, and dynamic dns name.

For example, say my current external ip address is 76.77.78.79 and my dynamic DNS name is test.ooguy.com.  If I access the Slingbox by http://76.77.78.79:8080/sb/dtv it works fine, but if I access it by http://test.ooguy.com:8080/sb/dtv there is a long delay and often does not work at all.  Logs for each look the same when the dns version works.

Once the Slingbox is running either http://test.ooguy.com:8080/Remote/dtv or http://76.77.78.79:8080/Remote/dtv works fine.

I am accessing the server with VLC.  The Windows version of VLC seems to work a little better with the dns name than my phone or Firestick, but all three have issues with the DNS name.

Often when I am testing this the server will get stuck.  If I go to the laptop and press Ctrl-C in the server log window once many sesssions will scroll by, saying "Starting remote stream 2", then 3, 4, 5, etc.

I cannot seem to get it running with MX Player.  I will post that issue and its logs in a separate thread.

I hope this info helps.  I really appreciate the effort.

Here is my config.ini
========================================================================
[SLINGBOXES]
sb1=dtv
sb2=tuner

[dtv]
sbtype="350/500/M1/M2"
password=<not shown>
ipaddress=192.168.1.69
port=5301
name=dtv
VideoSource = 1
Remote=remote_directv-h24.txt

[tuner]
sbtype="Solo/Pro/ProHD"
password=<not shown>
ipaddress=192.168.1.253
port=5001
name=tuner
VideoSource = 0

[SERVER]
port=8080
URLbase=sb
=========================================================================

Here is the log output where I access the Slingbox from my phone.  The phone is connected to the cellular network and not my local wifi.  I assume 172.56.10.83 and 172.56.11.73 were the IP addresses of my phone when I connected to the Slingbox server.  It is not my external home IP address.  I added comments with ***************
==============================================================================
C:\slingbox>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 c:
 REM use the full pathname to where your code is located
 cd \slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py config_m1.ini config_prohd.ini
)
Version : 3.08 Running on Windows-10-10.0.22621-SP0 pid= 11080
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL sb.
BOXES [('sb1', 'dtv'), ('sb2', 'tuner')]
Streamer Running:  10 config.ini tuner tuner 8080 8388608
Streamer Running:  10 config.ini dtv dtv 8080 8388608
Checking for slingbox at ('192.168.1.253', 5001)
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.69', 5301)
('192.168.1.253', 5001) OK
tuner Using slingbox at  ('192.168.1.253', 5001)
('192.168.1.69', 5301) OK
dtv Using slingbox at  ('192.168.1.69', 5301)
Streamer:  dtv Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Streamer:  tuner Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for dtv
Reading Custom Remote definition from remote_directv-h24.txt
Building page for tuner
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
***************** This is where I connect with my external home IP address ***************************
10/18/2022, 08:44:01   Streaming connection from ('172.56.10.83', 41860)
Starting remote stream 1

Slinginfo  "350/500/M1/M2" 12 30 192.168.1.69 5301 3072 10
dtv Starting Stream for  172.56.10.83:41860
Connecting... ('192.168.1.69', 5301) Control
dtv Selecting Video Source 1
Keycodes= [1, 2, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 25, 26, 27, 28, 29, 30, 31, 33, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 62, 63, 64, 65, 66, 67, 68, 69]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.69', 5301) Stream
dtv Stream started at 10/18/2022, 08:44:04  749 0
.10/18/2022, 08:44:23  dtv Stream Terminated for  172.56.10.83:41860 [WinError 10053] An established connection was aborted by the software in your host machine
0 active remote connections
dtv Shutting down connections
Streamer:  dtv Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
***************** This is where I connect using my dynamic DNS name *****************************
10/18/2022, 08:47:23   Streaming connection from ('172.56.11.73', 53479)
Starting remote stream 1

Slinginfo  "350/500/M1/M2" 12 30 192.168.1.69 5301 3072 10
dtv Starting Stream for  172.56.11.73:53479
Connecting... ('192.168.1.69', 5301) Control
dtv Selecting Video Source 1
Keycodes= [1, 2, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 25, 26, 27, 28, 29, 30, 31, 33, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 62, 63, 64, 65, 66, 67, 68, 69]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.69', 5301) Stream
dtv Stream started at 10/18/2022, 08:47:26  749 0
..10/18/2022, 08:47:46  dtv Stream Terminated for  172.56.11.73:53479 [WinError 10053] An established connection was aborted by the software in your host machine
0 active remote connections
dtv Shutting down connections
Streamer:  dtv Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
===============================================================================

Here is the log output when I do the following
- connect to my external home IP address http://76.77.78.79:8080/sb/dtv from a PC on my local LAN
- bring up http://test.ooguy.com:8080/Remote/dtv in Chrome from my Android phone on the celluar network
- press 8 on the remote page (this works fine)
- disconnect VLC
===============================================================================
10/18/2022, 09:29:23   Streaming connection from ('192.168.1.254', 50777)

Slinginfo  "350/500/M1/M2" 12 30 192.168.1.69 5301 3072 10
dtv Starting Stream for  192.168.1.254:50777
Connecting... ('192.168.1.69', 5301) Control
dtv Selecting Video Source 1
Keycodes= [1, 2, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 24, 25, 26, 27, 28, 29, 30, 31, 33, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 62, 63, 64, 65, 66, 67, 68, 69]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.69', 5301) Stream
dtv Stream started at 10/18/2022, 09:29:25  749 0
.10/18/2022, 09:29:37   RemoteControl connection from ('172.56.11.67', 34996)

Starting remote control stream handler for  ('172.56.11.67', 34996) to port 8081
Remote Control Connected
Exiting Remote Control Stream Handler for ('172.56.11.67', 34996)
10/18/2022, 09:29:41   RemoteControl connection from ('172.56.11.67', 60748)

Starting remote control stream handler for  ('172.56.11.67', 60748) to port 8081
Remote Control Connected
Button Clicked ImmutableMultiDict([('8', '16 : round'), ('Digits', '')]) 172.56.11.67
Exiting Remote Control Stream Handler for ('172.56.11.67', 60748)
dtv Got Streamer Control Message IR
dtv Sending IR keycode 16 1 for 172.56.11.67
.10/18/2022, 09:29:49  dtv Stream Terminated for  192.168.1.254:50777 [WinError 10053] An established connection was aborted by the software in your host machine
dtv Shutting down connections
Streamer:  dtv Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
==============================================================================

**Comment by GerryDazoo** [comment, marked as answer]
This is an issue with your DYN provider. It's taking a looong time to resolve your address. Once it's resolved the result gets cached locally and that's why the second access is quick. The local cache will timeout after awhile. You can verify this by clearing your DNS cache and then doing a ping to your dyndns name and see how long it takes. Sorry there's nothing I can do about this.

**Comment by tvpat** [comment]
I think I fixed the problem.

I am using using Dynu for my dynamic dns.  By default, it enables IPv6 for every entry you create.  I disabled IPv6 for that entry and now the stream starts immediately.  I re-tested it on an another entry where IPv6 was still enabled and it still had the problem.

Pat
